pkg/runtime/framework/plugins/mpi/mpi.go

```diff
@@ -85,12 +89,9 @@ func (m *MPI) Validate(_ client.Object, runtimeInfo *runtime.Info, _, newJobObj
 		return nil, allErrs
 	}
 
-	specPath := field.NewPath("spec")
-	if newJobObj.Spec.Trainer != nil && newJobObj.Spec.Trainer.NumProcPerNode != nil {
-		numProcPerNodePath := specPath.Child("trainer").Child("numProcPerNode")
-		numProcPerNode := *newJobObj.Spec.Trainer.NumProcPerNode
-		if numProcPerNode.Type != intstr.Int {
-			allErrs = append(allErrs, field.Invalid(numProcPerNodePath, newJobObj.Spec.Trainer.NumProcPerNode, "must have an int value"))
+	if trainJobTrainer := newJobObj.Spec.Trainer; trainJobTrainer != nil && trainJobTrainer.NumProcPerNode != nil {
+		if trainJobTrainer.NumProcPerNode.Type != intstr.Int {
+			allErrs = append(allErrs, field.Invalid(numProcPerNodePath, *trainJobTrainer.NumProcPerNode, "must have an int value"))
```

Comment:
Not strictly in the scope of that PR, but while we are at it, maybe the error message could be more specific, e.g. "must have an int value for MPI job".

Reply:
Done.